File: public/docs/_examples/toh-3/ts/app/hero-detail.component.ts
// #docplaster
// #docregion
// #docregion v1
import { Component, Input } from '@angular/core';

// #enddocregion v1
// #docregion hero-import
import { Hero } from './hero';
// #enddocregion hero-import

// #docregion v1
@Component({
  selector: 'my-hero-detail',
// #enddocregion v1
  // #docregion template
  template: `
    <div *ngIf="hero">
      <h3>{{hero.name}} details!</h3>
      <dl>
        <dt>id:</dt>
        <dd>{{hero.id}}</dd>
      </dl>
      <div>
        <label>name: 
          <input [(ngModel)]="hero.name" placeholder="name">
        </label>
      </div>
    </div>
  `
  // #enddocregion template
// #docregion v1
})
export class HeroDetailComponent {
// #enddocregion v1
// #docregion hero-input
  @Input()
// #docregion hero
  hero: Hero;
// #enddocregion hero
// #enddocregion hero-input
// #docregion v1
}
// #enddocregion v1

File: public/docs/_examples/toh-3/dart/lib/hero_detail_component.dart
// #docplaster
// #docregion
// #docregion v1
import 'package:angular2/core.dart';

// #enddocregion v1
// #docregion hero-import
import 'hero.dart';
// #enddocregion hero-import

// #docregion v1
@Component(
    selector: 'my-hero-detail',
    // #enddocregion v1
    // #docregion template
    template: '''
      <div *ngIf="hero != null">
        <h2>{{hero.name}} details!</h2>
        <div><label>id: </label>{{hero.id}}</div>
        <div>
          <label>name: </label>
          <input [(ngModel)]="hero.name" placeholder="name">
        </div>
      </div>'''
    // #enddocregion template
    // #docregion v1
    )
class HeroDetailComponent {
  // #enddocregion v1
  // #docregion inputs
  @Input()
  // #docregion hero
  Hero hero;
  // #enddocregion hero, inputs
  // #docregion v1
}
